File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/interceptor/ShortCircuitRedirectInterceptor.java
package com.roxiemobile.networkingapi.network.rest.interceptor;

import com.roxiemobile.networkingapi.network.rest.RestApiClient.HttpResponseException;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.Response;

public class ShortCircuitRedirectInterceptor extends AbstractRedirectInterceptor {

// MARK: - Methods

    /**
     * TODO
     */
    public @NotNull Response onRedirect(@NotNull Response response) throws IOException {
        // Throw an exception on redirects
        throw new HttpResponseException(decompressResponse(response));
    }
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/interceptor/ShortCircuitRedirectInterceptor.kt
package com.roxiemobile.networkingapi.network.rest.interceptor

import com.roxiemobile.networkingapi.network.rest.RestApiClient.HttpResponseException
import okhttp3.Response

class ShortCircuitRedirectInterceptor: AbstractRedirectInterceptor() {

// MARK: - Methods

    override fun onRedirect(response: Response): Response {
        // Throw an exception on redirects
        throw HttpResponseException(decompressResponse(response))
    }
}
